glib/src/collections/strv.rs

```diff
@@ -772,7 +772,7 @@ impl StrV {
     #[inline]
     pub fn extend_from_slice<S: AsRef<str>>(&mut self, other: &[S]) {
         // Nothing new to reserve as there's still enough space
-        if self.len + other.len() + 1 > self.capacity {
+        if other.len() >= self.capacity - self.len {
```

Comment:
Same issue in ptr_slice.rs and slice.rs.

And it might make sense to also change this for all the other calls to `reserve()`, e.g. in `insert()` just a few lines below